Assets/MRTK/SDK/Features/UX/Scripts/StateVisualizer/AnimationTarget.cs

```diff
@@ -0,0 +1,195 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License
+
+using Microsoft.MixedReality.Toolkit.Utilities;
+using System;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.UI.Interaction
+{
+    /// <summary>
+    /// Definition class for an animation target, utilized in the State Visualizer component.
+    /// </summary>
+    [Serializable]
+    public class AnimationTarget
+    {
+        [SerializeField]
+        [Tooltip("The target game object for animations.")]
+        private GameObject target;
+
+        /// <summary>
+        /// The target game object for animations.
+        /// </summary>
+        public GameObject Target
+        {
+            get => target;
+            set
+            {
+                if (IsTargetObjectValid(value))
+                {
+                    target = value;
+                }
+                else
+                {
+                    target = null;
+                    Debug.LogError("The Target property can only be a child of this game object, the target was set back to null");
```

Comment:
Should this set it to `null`, or just leave it as whatever it's currently set to (which might not be `null`!)?

Reply:
This property needs to be set to null because a target object that does not have a state visualizer attached or is a child of an object with state visualizer attached will never be valid.